### PR TITLE
Upgrade to gRPC 0.11.

### PR DESCRIPTION
**Pull request by arielshaqed**
Mostly this adds "NULL" to various reserved locations. But completion
queue handling is now much simpler -- really only need to read a
single success value for the entire grpc_call_start_batch() call.

First part of fixing #21 

Currently this is unable to use the trusted CAs list in Ubuntu Trusty.

### CODE REVIEW

**Comment by arielshaqed** [comment]
@yosih please review.

File: src/write_grpc.c [hunk 1]
```diff
@@ -397,42 +398,29 @@ static gpr_timespec get_deadline(cdtime_t delay)
  * accepted or cq is empty.  Returns a true value if all writes were
  * accepted.  Must call with cb->mutex held. */
 static int process_cq(grpc_callback *cb, int do_sleep) {
-  grpc_event *ev;
+  grpc_event ev;
 
   while (cb->write_accepted_counter < cb->write_counter) {
```
Comment (yosih) [inline src/write_grpc.c]: These 2 counters are unused

File: src/write_grpc.c [hunk 2]
```diff
@@ -582,18 +574,19 @@ static int do_flush_nolock(grpc_callback *cb, cdtime_t timeout) {
...
+  /* TODO(arielshaqed): grpc_raw_compressed_byte_buffer_create? */
+  byte_buffer = grpc_raw_byte_buffer_create(&slice_out, 1);
 
   num_ops = make_call_ops(
       byte_buffer, ops, sizeof ops / sizeof ops[0],
       &initial_metadata_recv, &response, &trailing_metadata_recv,
       &status, &status_details, &status_details_capacity);
   call = grpc_channel_create_call(
-      cb->channel,
+      cb->channel, NULL /* Parent call */, 0,
```
Comment (yosih) [inline src/write_grpc.c]: The extra parameters can go on separate lines

File: src/write_grpc.c [hunk 3]
```diff
@@ -810,80 +819,19 @@ static int write_data(const data_set_t *ds, const value_list_t *vl,
...
-      use_instance_credentials, service_account_json_filename);
-  grpc_credentials *cred_server = get_server_credentials(root_pem_filename);
-  grpc_credentials *ret = NULL;
-
-  if (cred_client && cred_server)
-    ret = grpc_composite_credentials_create(cred_server, cred_client);
-
-  if (cred_client) grpc_credentials_release(cred_client);
-  if (cred_server) grpc_credentials_release(cred_server);
+  if (service_account_json_filename != NULL)
```
Comment (yosih) [inline src/write_grpc.c:833]: Environment variables are the only way to pass these parameters? Specifically for GRPC_GOOGLE_CREDENTIALS_ENV_VAR, if it's not set, will it go through the metadata server?

**Comment by yosih** [comment]
Thanks

File: src/write_grpc.c (hunk 1 above)
Comment (arielshaqed) [inline src/write_grpc.c]: Removed them, and revamped the completion-queue wait and drain logic. PTAL at that, too (process_cq, drain_cq, and calls to them).

File: src/write_grpc.c (hunk 2 above)
Comment (arielshaqed) [inline src/write_grpc.c]: Done!

File: src/write_grpc.c (hunk 3 above)
Comment (arielshaqed) [inline src/write_grpc.c:833]: Yes, at least for this call :-(.

**tl;dr:** It is not clear how to compose an SSL server-side key and a JSON account client-side key. I discussed with Ido, he says to use the least amount of extra code. But AFAICT the library reads the environment variables _once_, and then reuses the values that it read. So if we have 2 `write_grpc` modules configured, `collectd` configuration is single-threaded and things should work.

**_Long...**_ The logic we have predates

**Comment by arielshaqed** [comment]
@yosih PTAL.

**Comment by arielshaqed** [comment]
Thanks!
